api/src/main/java/org/apache/iceberg/catalog/SessionCatalog.java

```diff
@@ -309,6 +309,17 @@ default List<Namespace> listNamespaces(SessionContext context) {
    */
   boolean dropNamespace(SessionContext context, Namespace namespace);
 
+  /**
+   * Drop a namespace. If the namespace exists and was dropped, this will return true.
+   *
+   * @param context session context
+   * @param namespace a {@link Namespace namespace}
+   * @param cascade – When true, deletes all objects under the namespace
```

Comment:
This needs a stricter definition IMO. In some instances below here you delete folders, sometimes you delete files, sometimes it deletes entries. We need a strong definition here.

Reply:
One big missing thing here is how this works with "purge"

Reply:
I think we have two options, drop_namespace with cascade will:
1) drop the tables not the data (no purge) 
2) drop the tables and delete the data (purge) 

We can standardize this in Iceberg and go with the preferred option, #1 sounds better as #2 can take a long time depending on number of tables and data in them. This behavior is not consistent across all catalogs, for example hive metastore cascade deletes all the data (purge) in tables where as others don't.

should I create a vote thread for this to decide on the preference or can we decide that in this PR? I can then update the implementation.

Reply:
@RussellSpitzer  documented the current purge behavior in the PR description and docs. 

Only Hive catalog purges the tables as that's the default behavior of hive when we pass the cascade=true parameter. 

I can make it consistent across all catalogs and modify current hive catalog behavior to just drop table without purge similar to other catalogs. Let me know your thoughts

Reply:
Yeah sorry I didn't get back to this but I do believe we probably need a discuss thread, or we can get feedback here but I think we need to hear from @jackye1995 and others who are maintaining other Catalog implementations.

Reply:
removed implementation for other catalogs, will initiate a discussion once api change in this PR is merged

Reply:
Glue by default does cascade drop if just call `glue.deleteDatabase`, but does not drop data in the table. So technically you can support that directly. But I am okay to publish another PR to support that, up to you. 

For doing purge or not, I think the behavior of cascade is not clearly specified in Spark. It could also be argued as something that is catalog-specific, just like the behavior of Hive and Glue are different and it will be difficult to satisfy the other behavior on both sides.

This is related to the issue about different behaviors of list namespace that comes up recently. Maybe we should make a page documenting all catalogs and their different behaviors. 

+1 to have a devlist discussion thread first.

Reply:
I will start a devlist discussion and create a one pager with current state. Thanks @jackye1995